src/main/java/com/google/sps/servlets/EditTaskServlet.java

```diff
@@ -39,67 +29,74 @@
 /** Servlet that handles the request for editing the details and rewards for a certain task. */
 @WebServlet("/tasks/edit")
 public class EditTaskServlet extends HttpServlet {
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        System.out.println("Modify");
-        String keyString = request.getParameter("task-id");
-        int rewardPts = getRewardingPoints(request, "edit-input");
-        if (rewardPts == -1) {
-            response.setContentType("text/html");
-            response.getWriter().println("Please enter a valid integer in the range of 0-200");
-            return;
-        }
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keyString = request.getParameter("task-id");
+    int rewardPts = getRewardingPoints(request, "reward-input");
+    if (rewardPts == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter a valid integer in the range of 0-200");
+      return;
+    }
 
-        // Get the task detail from the form input
-        String taskDetail = "";
-        String input = request.getParameter("edit-content-input");
-        if (input != null) {
-            taskDetail = input;
-        }
+    // Get the task detail from the form input
+    String taskDetail = "";
+    String input = request.getParameter("task-detail-input");
+    if (input != null) {
+      taskDetail = input;
+    }
 
-        // If the input is nonempty and valid, set the taskDetail value to the input value
-        if (!taskDetail.equals("")) {
-            Key taskKey = KeyFactory.stringToKey(keyString);
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            Entity task;
-            try {
-                task = datastore.get(taskKey);
-            } catch (EntityNotFoundException e) {
-                System.err.println("Unable to find the entity based on the input key");
-                response.sendError(HttpServletResponse.SC_NOT_FOUND, "The entity is not found in the database");
-                return;
-            }
+    // If the input is nonempty and valid, set the taskDetail value to the input value
+    if (!taskDetail.equals("")) {
```

Comment:
I think it's simpler and easier to follow this method if we return early (with an error?) when taskDetail is empty, like we do above if rewardPts is -1. Then you can dedent the rest of the method body to indicate that it's the normal flow of the method. Pseudocode:

```
...
if rewardPts == -1 {
  return error
}
...
if taskDetail.equals("") {
  return error
}
... do all the rest of the things ...

Reply:
Sure. I will change it in a new commit for both the `EditTaskServlet.java` and `TaskServlet.java` files.